storage.go

```diff
@@ -44,11 +52,68 @@ func (m *memoryStorage) Get(key string) (*RequestStatus, error) {
 // Complete sets a request to not be in progress, it is then determined to be
 // completed and that we should serve the result we got from a previous
 // request.
-func (m *memoryStorage) Complete(key string) error {
+func (m *memoryStorage) Complete(ctx context.Context, key string) error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
 	m.storage[key] = &RequestStatus{InProcess: false}
 
 	return nil
 }
+
+type redisStorage struct {
+	client *redis.Client
+	expiry time.Duration
+}
+
+const keyPrefix = "idemp:"
```

Comment:
i think we need to handle the prefixing in the same way as in the worker queue to handle the case where we share the redis instance in staging.

util.KubernetesNSSuffix(pwork.NAMESPACE, config.Env.PodNamespace)

Reply:
Good catch, I will make it configurable here since this is a generic package.